crates/parser/src/parser.rs

```diff
@@ -386,4 +388,15 @@ impl CompletedMarker {
     pub(crate) fn kind(&self) -> SyntaxKind {
         self.kind
     }
+
+    pub(crate) fn last_token(&self, p: &Parser<'_>) -> Option<SyntaxKind> {
+        let end_pos = self.end_pos as usize;
+        if end_pos > p.events.len() {
+            return None;
+        }
```

Comment:
```suggestion
        debug_assert_eq!(p.events[end_pos - 1], Event::Finish);
```
The early return should never happen here